src/gui/guiAnimatedImage.cpp

```diff
@@ -0,0 +1,85 @@
+#include "guiAnimatedImage.h"
+
+#include "client/guiscalingfilter.h"
+#include "client/tile.h" // ITextureSource
+#include "log.h"
+#include "porting.h"
+#include <string>
+
+GUIAnimatedImage::GUIAnimatedImage(gui::IGUIEnvironment *env, gui::IGUIElement *parent,
+		s32 id, const core::rect<s32> &rectangle, const std::string &name,
+		ISimpleTextureSource *tsrc) :
+		gui::IGUIElement(gui::EGUIET_ELEMENT, env, parent, id, rectangle),
+		m_name(name), m_tsrc(tsrc), m_texture(nullptr), m_global_time(0),
+		m_frame_idx(0), m_frame_count(1), m_frame_duration(1), m_frame_time(0)
+{
+	// Expected format: "texture_name:frame_count,frame_duration"
+	// If this format is not met, the string will be loaded as a normal texture
+
+	m_frame_idx = 0;
+
+	std::string::size_type colon_position = name.find(':', 0);
+	std::string::size_type comma_position = name.find(',', 0);
+
+	if (comma_position > colon_position && comma_position < name.size()) {
+		m_texture = m_tsrc->getTexture(name.substr(0, colon_position));
+
+		m_frame_count = std::max(stoi(name.substr(
+					colon_position + 1, comma_position - colon_position - 1)), 1);
+
+		m_frame_duration = std::max(stoi(name.substr(comma_position + 1)), 1);
+	} else {
+		// Leave the count/duration and display a static image
+		m_texture = m_tsrc->getTexture(name);
```

Comment:
I'd output an error here. It's clearly a wrong format for this formspec element.

Reply:
Done, I think the message is good but feel free to re-check now!

Reply:
Good so far, but the line needs breaking (>80 characters). It also wouldn't hurt to include a hint about the error source, example: `animated_image[]: Invalid texture ...`. I know the element name does not fit into this class but it'll be helpful to modders.

Reply:
Looks like you went and fixed it for me, thank you!
Does that mean this conversation can be resolved?

Reply:
Yes. If requested changes were addressed feel free to mark my comments as resolved.